superlist_attribute.go

```diff
@@ -167,3 +167,148 @@ func (s SuperListAttribute) GetDataSource(ctx context.Context) schemaD.Attribute
 	a.MarkdownDescription = genDataSourceAttrDescription(ctx, a.MarkdownDescription, deprecationMessage, a.Validators)
 	return a
 }
+
+// * SuperTypeOf
+
+var _ Attribute = SuperListAttributeOf[struct{}]{}
+
+type SuperListAttributeOf[T any] struct {
+	Deprecated *Deprecated
+	Common     *schemaR.ListAttribute
+	Resource   *schemaR.ListAttribute
+	DataSource *schemaD.ListAttribute
+}
+
+// IsResource returns true if the attribute is a resource attribute.
+func (s SuperListAttributeOf[T]) IsResource() bool {
+	return s.Resource != nil || s.Common != nil
+}
+
+// IsDataSource returns true if the attribute is a data source attribute.
+func (s SuperListAttributeOf[T]) IsDataSource() bool {
+	return s.DataSource != nil || s.Common != nil
+}
+
+//nolint:dupl
+func (s SuperListAttributeOf[T]) GetResource(ctx context.Context) schemaR.Attribute {
+	var (
+		common   schemaR.ListAttribute
+		resource schemaR.ListAttribute
+	)
+
+	if s.Common != nil {
+		common = *s.Common
+	}
+
+	if s.Resource != nil {
+		resource = *s.Resource
+	}
+
+	a := schemaR.ListAttribute{
+		Required:            computeIsRequired(common, resource),
+		Optional:            computeIsOptional(common, resource),
+		Computed:            computeIsComputed(common, resource),
+		Sensitive:           computeIsSensitive(common, resource),
+		MarkdownDescription: computeMarkdownDescription(common, resource),
+		Description:         computeDescription(common, resource),
+		DeprecationMessage:  computeDeprecationMessage(common, resource),
+		ElementType:         common.ElementType,
+	}
+
+	a.Validators = append(a.Validators, common.Validators...)
+	a.Validators = append(a.Validators, resource.Validators...)
+	a.PlanModifiers = append(a.PlanModifiers, common.PlanModifiers...)
+	a.PlanModifiers = append(a.PlanModifiers, resource.PlanModifiers...)
+
+	defaultVDescription := ""
+
+	if s.Common != nil {
```

Comment:
you can by pass this action with : common var

if common.CustomType != nil {
			a.CustomType = common.CustomType
		}